staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -310,6 +330,21 @@ func (t *tracker) add(gvr schema.GroupVersionResource, obj runtime.Object, ns st
 
 	t.objects[gvr] = append(t.objects[gvr], obj)
 
+	if ns != "" && t.watchers[gvr] != nil && t.watchers[gvr][""] != nil {
```

Comment:
make a helper method that takes the watchers map and a namespace and returns a slice of the watchers to send the even to.  This block will exist in the delete method too.